### PR TITLE
Improve Error Messages + Refactor Switch E2E Tests

### PR DESCRIPTION
**Pull request by lawrencewin**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [x] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Currently, for most developers, it is difficult to figure out what the root issue is when a FURN E2E test fails because of opaque error messages. The existing E2E tests are also in need of some clean up in terms of code and PageObject design.

This commit is part of a larger effort to address the above. Here, I've updated error messages, use new BasePage methods, and did general cleanup for the following tests:

- Switch

### Verification

Tests pass locally and in CI.

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: apps/E2E/src/Switch/pages/SwitchPageObject.ts [hunk 1]
```diff
@@ -18,48 +12,30 @@ class SwitchPageObject extends BasePage {
...
-    });
-  }
-
-  async toggleSwitchToUnchecked(): Promise<void> {
-    if (await this.isSwitchChecked()) {
+  async setSwitchState(toggleState: boolean) {
+    const currState = await this.isSwitchChecked();
+    if (toggleState !== currState) {
       await (await this._primaryComponent).click();
+      await this.waitForCondition(
```
Comment (samuelfreiberg) [inline apps/E2E/src/Switch/pages/SwitchPageObject.ts:19]: Seems redundant with `waitForSwitchStateChange()`

File: apps/E2E/src/Switch/specs/Switch.spec.win.ts [hunk 2]
```diff
@@ -28,18 +27,31 @@ describe('Switch Accessibility Testing', () => {
     await SwitchPageObject.scrollToTestElement();
   });
 
-  it('Switch - Validate accessibilityRole is correct', async () => {
-    await expect(await SwitchPageObject.getAccessibilityRole()).toEqual(BUTTON_A11Y_ROLE);
+  it('Validate "accessibilityRole" defaults to Button "ControlType" element attribute.', async () => {
```
Comment (samuelfreiberg) [inline apps/E2E/src/Switch/specs/Switch.spec.win.ts:30]: The "Button" part is a bit confusing. What do you mean by that?

File: apps/E2E/src/Switch/specs/Switch.spec.win.ts [hunk 3]
```diff
@@ -48,67 +60,97 @@ describe('Switch Functional Testing', () => {
...
-    await SwitchPageObject.clickComponent();
-    await SwitchPageObject.waitForSwitchChecked(PAGE_TIMEOUT);
-
-    await expect(await SwitchPageObject.didOnChangeCallbackFire()).toBeTruthy();
+    await SwitchPageObject.click(SwitchPageObject._primaryComponent);
 
     /* Validate the Switch is toggled ON */
-    await expect(await SwitchPageObject.isSwitchChecked()).toBeTruthy();
+    await expect(
+      await SwitchPageObject.waitForSwitchStateChange(true, 'Clicked the primary switch to turn it on, but it remained on.'),
```
Comment (samuelfreiberg) [inline apps/E2E/src/Switch/specs/Switch.spec.win.ts:78]: "But it remained **off**"

File: apps/E2E/src/Switch/specs/Switch.spec.win.ts (hunk 2 above)
Comment (lawrencewin) [reply apps/E2E/src/Switch/specs/Switch.spec.win.ts:30]: For this test, I'm trying to say that the ControlType element attribute should be equal to a "Button" type - the value passed into accessibilityRole is "ControlType.Button"

**Review by samuelfreiberg — APPROVED** [review]

**Pull request merged**